mktl/mktlcoms.py

```
@@ -451,58 +451,6 @@ def _handle_control_message(self, msg):
-
-        Encodes the identity, address, and known keys into a StoreConfig payload
-        and sends it as a `set` to `registry.config`.
-        """
-        logger = getLogger(__name__)
-        keys = list(self.authoritative_keys.keys())
-        user_keys = list(keys)
-        user_keys.remove(f'{self.identity}.mktl_control')
-        if not user_keys:
-            logger.debug("No authoritative keys to announce (besides mktl_control), will not register.")
-            return
-        logger.debug("Attempting to send registry config...")
-
-        payload = {'identity': self.identity,
-                   'address': self._bind_address,
-                   'keys': keys,
-                   }
-
-        self.set('registry.config', payload)
-
-    def _query_registry_owner(self, key: str) -> Optional[Tuple[str, str]]:
-        """
-        Query the registry for the owner of a specific key.
-
-        Sends a `get` request to `registry.owner` and expects a response
-        with `identity` and `address`.
-
-        Args:
-            key: The key to resolve.
-
-        Returns:
-            Tuple of (identity, address), or raises on failure.
-        """
-        logger = getLogger(__name__)
-        if not self.registry_addr:
-            logger.warning('Registry address not configured.')
-            return None
-
-        result = self.get('registry.owner', {'key': key})
-        identity = result.json_data.get('identity')
-        address = result.json_data.get('address')
-        if identity and address:
-            self._routing_table[key] = (identity, address)
-            logger.info(f"Registry resolution for {key}: identity={identity}, address={address}")
-            return identity, address
-        else:
-            logger.warning(f"Could not resolve owner for key={key}")
-            return None
-
     def _resolve_destination(self, key: str, destination: Optional[str]) -> str:
```

I'd be inclined to axe this entirely and fail fast in get/set. The destination argument there is now simply a direct override to the routing table.